### ISSUE TITLE
Low Attack Performance of PSO attack

### ISSUE THREAD

**Issue by lethaiq** [QData/TextAttack#374]
I run PSO attack through provided pre-trained `wordCNN` and `wordLSTM` but the attack success rate is very low. Is this because the PSO attack is weaker than other attacks? I wonder you observe a different behavior from me?

**Comment by jinyongyoo** [comment]
Can you provide the exact attack you ran (e.g. recipe or other parameters)?

**Comment by lethaiq** [comment]
@jinyongyoo `textattack attack --model cnn-mr --recipe pso --num-examples 100`
Output:

`+-------------------------------+--------+
| Attack Results                |        |
+-------------------------------+--------+
| Number of successful attacks: | 0      |
| Number of failed attacks:     | 77     |
| Number of skipped attacks:    | 23     |
| Original accuracy:            | 77.0%  |
| Accuracy under attack:        | 77.0%  |
| Attack success rate:          | 0.0%   |
| Average perturbed word %:     | nan%   |
| Average num. words per input: | 17.98  |
| Avg num queries:              | 1201.0 |
+-------------------------------+--------+
textattack: Attack time: 36.45260047912598s`

**Comment by jxmorris12** [comment]
@jind11 does a 0% attack success rate seem correct to you?

I think some of our CNN models are somewhat overfitted. This could be a factor.

**Comment by lethaiq** [comment]
@jxmorris12 I also run this attack on a custom model I trained externally and the attack success rate is very low. I don’t have time this week but I could look at this more closely next month if it is not resolved by then.

**Comment by jinyongyoo** [comment]
It's because of a bug with WordSwapHowNet. PR #382 should fix it.

**Comment by lethaiq** [comment]
> It's because of a bug with WordSwapHowNet. PR #382 should fix it.

Great Thanks
